### PR TITLE
Chemmaster Escape Case

### PR DESCRIPTION
**Pull request by Kurfursten**
This adds a special escape case to the Chemmaster. If you realize that you accidentally just asked for 20 empty pills, you can cancel by calling your pills 'Cancel'.

### CODE REVIEW

**Comment by Probe1** [comment]
that's not a useless feature it's a rather useful one that's more intuitive.

**Comment by SkowronX** [comment]
This should have been a thing in the first place.

File: code/modules/reagents/Chemistry-Machinery.dm [hunk 1]
```diff
@@ -472,7 +472,8 @@ USE THIS CHEMISTRY DISPENSER FOR MAPS SO THEY START AT 100 ENERGY
 			if (!count) return
 			var/amount_per_pill = reagents.total_volume/count
 			if (amount_per_pill > 50) amount_per_pill = 50
-			var/name = reject_bad_text(input(usr,"Name:","Name your pill!","[reagents.get_master_reagent_name()] ([amount_per_pill] units)"))
+			var/name = reject_bad_text(input(usr,"Name:","Name your pill! ('Cancel' exits)","[reagents.get_master_reagent_name()] ([amount_per_pill] units)"))
```
Comment (d3athrow) [inline code/modules/reagents/Chemistry-Machinery.dm]:
~~~
var/name = input(stuff) as null|text
if(!name) return
~~~

**Comment by Shaska-** [comment]
Can you make it cancel if you don't give them a name as well?
Bit easier to remember, and that way we don't end up with 20 " pill"s

**Comment by Kurfursten** [comment]
D3athrow and Shaska both pose good points. I'll rewrite it to use blank text instead of 'Cancel' (and the box reminder will inform players of this as well)

**Comment by d3athrow** [comment]
> I'll rewrite it to use blank text instead of 'Cancel'

Yes what Nexis said and my comment, if you use input(thing) as null|something it gives you a cancel button.

**Comment by d3athrow** [comment]
@Kurfursten make the change I suggested and this can be merged.

**Pull request merged**